Streetcode/Streetcode.BLL/Middleware/ApiRequestResponseMiddleware.cs

```diff
@@ -0,0 +1,142 @@
+        }
+
+        private void LogResponse(string response)
+        {
+            _loggerService.LogInformation(response);
+        }
+
+        private async Task<string> GetRequestAsTextAsync(HttpRequest request)
+        {
+            var requestBody = request.Body;
+
+            request.EnableBuffering();
+
+            var buffer = new byte[Convert.ToInt32(request.ContentLength)];
+
+            await request.Body.ReadAsync(buffer, 0, buffer.Length);
+
+            var bodyAsText = Encoding.UTF8.GetString(buffer);
+
+            request.Body = requestBody;
+
+            return $"{request.Scheme} {request.Host} {request.Path} {request.QueryString} {bodyAsText}";
+        }
+
+        private static async Task<string> GetResponseAsTextAsync(HttpResponse response)
+        {
+            response.Body.Seek(0, SeekOrigin.Begin);
+
+            var encoding = Encoding.UTF8;
+
+            var text = await new StreamReader(response.Body, encoding).ReadToEndAsync();
+
+            response.Body.Seek(0, SeekOrigin.Begin);
+
+            return text;
+        }
+
+        private string GetFilteredResponse(string response)
+        {
+            try
+            {
+                var jsonObject = JToken.Parse(response);
+
+                if (jsonObject.Type == JTokenType.Array)
+                {
+                    var jsonArray = (JArray)jsonObject;
+
+                    foreach (var item in jsonArray)
+                    {
+                        TruncateProperties(item);
+                    }
+                }
+                else if (jsonObject.Type == JTokenType.Object)
+                {
+                    TruncateProperties(jsonObject);
+                }
+
+                return jsonObject.ToString();
+            }
+            catch (Exception)
+            {
+                return null;
+            }
+        }
+
+        private void TruncateProperties(JToken token)
+        {
+            foreach (var property in token.Children<JProperty>().ToList())
+            {
+                if (!_options.PropertiesToIgnore.Contains(property.Name.ToString().ToLower()))
+                {
+                    if (!_options.PropertiesToShorten.Contains(property.Name.ToString().ToLower()))
+                    {
+                        continue;
+                    }
+
+                    if (property.Value.Type == JTokenType.String && property.Value.ToString().Length > _options.MaxResponseLength)
+                    {
+                        property.Value = new JValue(property.Value.ToString().Substring(0, _options.MaxResponseLength));
+                    }
+                    else if (property.Value.Type == JTokenType.Object || property.Value.Type == JTokenType.Array)
+                    {
+                        TruncateProperties(property.Value);
+                    }
+                }
+                else
+                {
+                    property.Remove();
+                }
```

Comment:
Can we invert this first if? It will make it easier to read
```suggestion
                if (_options.PropertiesToIgnore.Contains(property.Name.ToString().ToLower()))
                {
                    property.Remove();
                    continue;
                }
                if (!_options.PropertiesToShorten.Contains(property.Name.ToString().ToLower()))
                {
                    continue;
                }

                 if (property.Value.Type == JTokenType.String && property.Value.ToString().Length > _options.MaxResponseLength)
                {
                    property.Value = new JValue(property.Value.ToString().Substring(0, _options.MaxResponseLength));
                }
                else if (property.Value.Type == JTokenType.Object || property.Value.Type == JTokenType.Array)
                {
                    TruncateProperties(property.Value);
                }
            }
```